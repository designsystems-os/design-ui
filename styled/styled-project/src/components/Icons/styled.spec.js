import React from 'react';
import TestRenderer from 'react-test-renderer';
import 'jest-styled-components';

import Icon from './styled';

////////////////////////////////////////
/* Your Styled Icon Tests Go Here...  */
////////////////////////////////////////

/*
 * Default Styles
 */
test('it renders default styles', () => {
  const component = TestRenderer.create(<Icon />).toJSON();
  expect(component).toMatchSnapshot();
  expect(component).toHaveStyleRule('height', '40px');
  expect(component).toHaveStyleRule('width', '40px');
});

/*
 * Custom Styles
 */
test('it renders custom styles', () => {
  const component = TestRenderer.create(
    <Icon height="28px" width="28px" />,
  ).toJSON();
  expect(component).toMatchSnapshot();
  expect(component).toHaveStyleRule('height', '28px');
  expect(component).toHaveStyleRule('width', '28px');
});
