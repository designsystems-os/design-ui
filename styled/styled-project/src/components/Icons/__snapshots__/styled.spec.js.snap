// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders custom styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 28px;
  width: 28px;
  border-radius: 50%;
  cursor: pointer;
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0:hover,
.c0:focus,
.c0:active {
  background: rgba(0,0,0,0.1);
  color: #000;
}

<div
  className="c0"
  display="flex"
  height="28px"
  width="28px"
/>
`;

exports[`it renders default styles 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 40px;
  width: 40px;
  border-radius: 50%;
  cursor: pointer;
  -webkit-transition: 200ms ease-out;
  transition: 200ms ease-out;
}

.c0:hover,
.c0:focus,
.c0:active {
  background: rgba(0,0,0,0.1);
  color: #000;
}

<div
  className="c0"
  display="flex"
  height="40px"
  width="40px"
/>
`;
